extensions/cloud/portability-cloud-microsoft/src/main/java/org/dataportabilityproject/cloud/microsoft/cosmos/MicrosoftCloudExtension.java

```diff
@@ -8,45 +8,116 @@
 
 import java.util.Objects;
 
+import static java.lang.System.getenv;
+
 /**
- * Bootstraps the Microsoft cloud extension.
+ * Bootstraps the Microsoft cloud extension. The extension uses Azure Key Vault for storing secrets
+ * and Azure Cosmos DB and Blob Storage for persistence.
+ *
+ * <p>During initialization, this extension first establishes a connection to the Azure Key Vault to
+ * obtain the account key secret for accessing Cosmos DB. As the configuration for connecting to the
+ * vault is sensitive, it is expect that the runtime process orchestrator will provide it via
+ * environment variables in a secure manner.
+ *
+ * <p>The following configuration values (environment or runtime configuration) are required:
+ *
+ * <ul>
+ *   <li>{@link #AZURE_VAULT_NAME}: The name of the Azure Vault instance
+ *   <li>{@link #AZURE_VAULT_TENANT_ID}: The name of the Active Directory tenant
+ *   <li>{@link #AZURE_VAULT_CLIENT_ID}: The id of the data portability app registered in Azure
+ *       Active Directory
+ *   <li>{@link #AZURE_VAULT_CLIENT_SECRET}: A secret set in Active Directory as the data
+ *       portability app password
+ *   <li>{@link #AZURE_COSMOS_ACCOUNT_NAME}: The account name of the Azure Cosmos DB instance.
+ *   <li>{@link #AZURE_COSMOS_PARTITION_KEY}: Optional. Used for selecting a Cosmos DB partition if
+ *       desired.
+ * </ul>
+ *
+ * Azure Setup.
  *
- * <p>
+ * <p>The following services must be setup on Azure:
+ *
+ * <ul>
+ *   <li>Active Directory: Active Directory must be enabled with a tenant
```

Comment:
FYI for the GCP stuff, we scripted this to make it easier for someone to follow all the setup steps

(See config/gcp/setup_gke_environment.sh)

Though I don't think config/ is the best place for it any more. I'm going to look into moving GCP setup stuff into the google cloud extension, and move anything specific to our old GCP test project into a demo-google-deployment distribution (#202)

Reply:
OK. I was planning on having the final config put into the documentation as well. It was a chore figuring out how to configure the gazillion parts of Azure  :-0

Reply:
Azure has a CLI that even runs in the browser, so I'll have to try something like what you said once all of the moving parts get organized.